[zh] translate RBAC good practices (#34128)
kubernetes/website: content/zh/docs/concepts/security/rbac-good-practices.md

<!--
### Workload creation

Users who are able to create workloads (either Pods, or
[workload resources](/docs/concepts/workloads/controllers/) that manage Pods) will
be able to gain access to the underlying node unless restrictions based on the Kubernetes
[Pod Security Standards](/docs/concepts/security/pod-security-standards/) are in place.
-->
### 工作负载的创建 {#workload-creation}

能够创建工作负载的用户（Pod 或管理 Pod 的[工作负载资源](/zh/docs/concepts/workloads/controllers/)） 
能够访问下层的节点，除非基于 Kubernetes 的
[Pod 安全标准](/zh/docs/concepts/security/pod-security-standards/)做限制。

<!--
Users who can run privileged Pods can use that access to gain node access and potentially to
further elevate their privileges. Where you do not fully trust a user or other principal
with the ability to create suitably secure and isolated Pods, you should enforce either the
**Baseline** or **Restricted** Pod Security Standard.
You can use [Pod Security admission](/docs/concepts/security/pod-security-admission/)
or other (third party) mechanisms to implement that enforcement.
-->
可以运行特权 Pod 的用户可以利用该访问权限获得节点访问权限，
并可能进一步提升他们的特权。如果你不完全信任某用户或其他主体，
不相信他们能够创建比较安全且相互隔离的 Pod，你应该强制实施 **Baseline**
或 **Restricted** Pod 安全标准。

> 你可以使用[Pod 安全性准入](/zh/docs/concepts/security/pod-security-admission/)或其他（第三方）机制来强制实施这些限制。

Reviewer: ```suggestion
你可以使用 [Pod 安全性准入](/zh/docs/concepts/security/pod-security-admission/)或其他（第三方）机制来强制实施这些限制。
```

in-built protections against privilege escalation, allowing users to create bindings to 
roles with rights they do not already have.
-->
### bind 动词  {#bind-verb}

与 `escalate` 动作类似，授予此权限的用户可以绕过 Kubernetes
对权限提升的内置保护，用户可以创建并绑定尚不具有的权限的角色。

<!--
### Impersonate verb

This verb allows users to impersonate and gain the rights of other users in the cluster. 
Care should be taken when granting it, to ensure that excessive permissions cannot be gained 
via one of the impersonated accounts.
-->
### impersonate 动词 {#impersonate-verb}

此动词允许用户伪装并获得集群中其他用户的权限。
授予它时应小心，以确保通过其中一个伪装账号不会获得过多的权限。

<!--
### CSRs and certificate issuing

The CSR API allows for users with `create` rights to CSRs and `update` rights on `certificatesigningrequests/approval` 
where the signer is `kubernetes.io/kube-apiserver-client` to create new client certificates 
which allow users to authenticate to the cluster. Those client certificates can have arbitrary 
names including duplicates of Kubernetes system components. This will effectively allow for privilege escalation.
-->
### CSR 和证书颁发 {#csrs-and-certificate-issuing}

CSR API 允许用户拥有 `create` CSR 的权限和 `update`
`certificatesigningrequests/approval` 的权限，

> 其中签名者是 `kubernetes.io/kube-apiserver-client`,

Reviewer: ```suggestion
其中签名者是 `kubernetes.io/kube-apiserver-client`，
```

mutating webhooks, also mutate admitted objects.
-->
###  控制准入 Webhook {#control-admission-webhooks}

可以控制 `validatingwebhookconfigurations` 或 `mutatingwebhookconfigurations`
的用户可以控制能读取任何允许进入集群的对象的 webhook，
并且在有变更 webhook 的情况下，还可以变更准入的对象。

<!--
## Kubernetes RBAC - denial of service risks {#denial-of-service-risks}

### Object creation denial-of-service {#object-creation-dos}
Users who have rights to create objects in a cluster may be able to create sufficient large 
objects to create a denial of service condition either based on the size or number of objects, as discussed in
[etcd used by Kubernetes is vulnerable to OOM attack](https://github.com/kubernetes/kubernetes/issues/107325). This may be
specifically relevant in multi-tenant clusters if semi-trusted or untrusted users 
are allowed limited access to a system.

One option for mitigation of this issue would be to use [resource quotas](/docs/concepts/policy/resource-quotas/#object-count-quota)
to limit the quantity of objects which can be created.
-->
## Kubernetes RBAC - 拒绝服务攻击的风险 {#denial-of-service-risks}

### 对象创建拒绝服务 {#object-creation-dos}
有权在集群中创建对象的用户根据创建对象的大小和数量可能会创建足够大的对象，

> 产生拒绝服务状况，如[Kubernetes 使用的 etcd 容易受到 OOM 攻击]（https://github.com/kubernetes/kubernetes/issues/107325）中的讨论。

Reviewer: Link should use `[]()` instead of `[]（）`.
Missing a space.
```suggestion
产生拒绝服务状况，如 [Kubernetes 使用的 etcd 容易受到 OOM 攻击](https://github.com/kubernetes/kubernetes/issues/107325) 中的讨论。
```

-->
在命名空间中创建工作负载还会授予对该命名空间中 Secret 的间接访问权限。
在 kube-system 或类似特权的命名空间中创建 Pod 
可以授予用户不需要通过 RBAC 即可获取 Secret 访问权限。

<!--
### Persistent volume creation

As noted in the [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems) documentation, access to create PersistentVolumes can allow for escalation of access to the underlying host. Where access to persistent storage is required trusted administrators should create 
PersistentVolumes, and constrained users should use PersistentVolumeClaims to access that storage.
-->
### 持久卷的创建 {#persistent-volume-creation}

如 [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems) 
文档中所述，创建 PersistentVolumes 的权限可以提权访问底层主机。
如果需要访问 PersistentVolume，受信任的管理员创建 `PersistentVolume`，
受约束的用户使用 `PersistentVolumeClaim` 访问该存储。

<!--
### Access to `proxy` subresource of Nodes

Users with access to the proxy sub-resource of node objects have rights to the Kubelet API, 
which allows for command execution on every pod on the node(s) which they have rights to. 
This access bypasses audit logging and admission control, so care should be taken before 
granting rights to this resource.
-->

> ### 访问节点的 `proxy` 子资源  {#access-to-proxy-subresource-of-nodes}

Reviewer 2: ```suggestion
### 访问 Node 的 `proxy` 子资源  {#access-to-proxy-subresource-of-nodes}
```

> 有权访问节点对象的代理子资源的用户有权访问 Kubelet API，

Reviewer 2: ```suggestion
有权访问 Node 对象的 proxy 子资源的用户有权访问 Kubelet API，
```

> 这允许在他们有权访问的节点上的每个 Pod 上执行命令。

Reviewer 2: ```suggestion
这允许在他们有权访问的节点上的所有 Pod 上执行命令。
```

此访问绕过审计日志记录和准入控制，因此在授予对此资源的权限前应小心。

<!--
### Escalate verb

Generally the RBAC system prevents users from creating clusterroles with more rights than 
they possess. The exception to this is the `escalate` verb. As noted in the [RBAC documentation](/docs/reference/access-authn-authz/rbac/#restrictions-on-role-creation-or-update),
users with this right can effectively escalate their privileges.
-->

> ### 提权动作 {#escalate-verb}

Reviewer 2: ```suggestion
### esclate 动词   {#escalate-verb}
```

通常，RBAC 系统会阻止用户创建比他所拥有的更多权限的 `ClusterRole`。

> 而 `escalate` 动作是个例外。如

Reviewer 2: ```suggestion
而 `escalate` 动词是个例外。如
```

[RBAC 文档](/zh/docs/reference/access-authn-authz/rbac/#restrictions-on-role-creation-or-update)
中所述，拥有此权限的用户可以有效地提升他们的权限。

<!--
### Bind verb

Similar to the `escalate` verb, granting users this right allows for bypass of Kubernetes 
in-built protections against privilege escalation, allowing users to create bindings to 
roles with rights they do not already have.
-->

> ### 绑定动作  {#bind-verb}

Reviewer 2: ```suggestion
### bind 动词    {#bind-verb}
```

与 `escalate` 动作类似，授予此权限的用户可以绕过 Kubernetes
对权限提升的内置保护，用户可以创建并绑定尚不具有的权限的角色。

<!--
### Impersonate verb

This verb allows users to impersonate and gain the rights of other users in the cluster. 
Care should be taken when granting it, to ensure that excessive permissions cannot be gained 
via one of the impersonated accounts.
-->

> ### 伪装动作 {#impersonate-verb}

Reviewer 2: ```suggestion
### impersonate 动词  {#impersonate-verb}
```

> 此动作允许用户模拟并获得集群中其他用户的权限。

Reviewer 2: ```suggestion
此动词允许用户伪装并获得集群中其他用户的权限。
```

> 授予它时应小心，以确保通过其中一个模拟账号不会获得过多的权限。

Reviewer 2: ```suggestion
授予它时应小心，以确保通过其中一个伪装账号不会获得过多的权限。
```

<!--
### CSRs and certificate issuing

The CSR API allows for users with `create` rights to CSRs and `update` rights on `certificatesigningrequests/approval` 
where the signer is `kubernetes.io/kube-apiserver-client` to create new client certificates 
which allow users to authenticate to the cluster. Those client certificates can have arbitrary 
names including duplicates of Kubernetes system components. This will effectively allow for privilege escalation.
-->
### CSR 和证书颁发 {#csrs-and-certificate-issuing}

CSR API 允许用户拥有 `create` CSR 的权限和 `update`
`certificatesigningrequests/approval` 的权限，
其中签名者是 `kubernetes.io/kube-apiserver-client`,
通过此签名创建的客户端证书允许用户向集群进行身份验证。
这些客户端证书可以包含任意的名称，包括 Kubernetes 系统组件的副本。

> 这将有利于特权升级。

Reviewer 2: ```suggestion
这将有利于特权提级。
```

<!--
### Token request

Users with `create` rights on `serviceaccounts/token` can create TokenRequests to issue 
tokens for existing service accounts. 
-->
### 令牌请求 {#token-request}

拥有 `serviceaccounts/token` 的 `create` 权限的用户可以创建 

> TokenRequests 来发布现有服务帐户的令牌。

Reviewer 2: ```suggestion
TokenRequest 来为现有服务帐户发放令牌。
```

mutating webhooks, also mutate admitted objects.
-->
###  控制准入 Webhook {#control-admission-webhooks}

可以操作 `validatingwebhookconfigurations` 或 `mutatingwebhookconfigurations`
的用户可以控制能读取任何允许进入集群的对象的 webhook，
并且在有变更 webhook 的情况下，还可以变更准入的对象。

<!--
## Kubernetes RBAC - denial of service risks {#denial-of-service-risks}

### Object creation denial-of-service {#object-creation-dos}
Users who have rights to create objects in a cluster may be able to create sufficient large 
objects to create a denial of service condition either based on the size or number of objects, as discussed in
[etcd used by Kubernetes is vulnerable to OOM attack](https://github.com/kubernetes/kubernetes/issues/107325). This may be
specifically relevant in multi-tenant clusters if semi-trusted or untrusted users 
are allowed limited access to a system.

One option for mitigation of this issue would be to use [resource quotas](/docs/concepts/policy/resource-quotas/#object-count-quota)
to limit the quantity of objects which can be created.
-->
## Kubernetes RBAC - 拒绝服务攻击的风险 {#denial-of-service-risks}

### 对象创建拒绝服务 {#object-creation-dos}
有权在集群中创建对象的用户根据创建对象的大小和数量可能会创建足够大的对象，

> 创造出拒绝服务攻击的条件，如[Kubernetes 使用的 etcd 容易受到 OOM 攻击]（https://github.com/kubernetes/kubernetes/issues/107325）中的讨论。

Reviewer 2: ```suggestion
产生拒绝服务状况，如 [Kubernetes 使用的 etcd 容易受到 OOM 攻击]（https://github.com/kubernetes/kubernetes/issues/107325）中的讨论。
```

Within Kubernetes RBAC there are a number of privileges which, if granted, can allow a user or a service account
to escalate their privileges in the cluster or affect systems outside the cluster.

This section is intended to provide visibility of the areas where cluster operators 
should take care, to ensure that they do not inadvertantly allow for more access to clusters than intended.
-->
## Kubernetes RBAC - 特权提权的风险 {#privilege-escalation-risks}

在 Kubernetes RBAC 中有许多特权，如果被授予，
用户或服务帐户可以提升其在集群中的权限并可能影响集群外的系统。

本节旨在为集群操作员提供应注意所在区域的可见性，
以确保它们不会无意中允许对集群的访问超出预期。

<!--
### Listing secrets

It is generally clear that allowing `get` access on Secrets will allow a user to read their contents.
It is also important to note that `list` and `watch` access also effectively allow for users to reveal the Secret contents.
For example, when a List response is returned (for example, via `kubectl get secrets -A -o yaml`), the response
includes the contents of all Secrets.
-->
### 列举 Secret {#listing-secrets}

通常很清楚，若设置 Secrets 的 `get` 访问权限，
用户可以获取其内容。 同样重要需要注意的是：`list`和
`watch` 访问也允许用户获取 Secret 的内容。
例如，当返回 List 响应时（例如,通过 

> `kubectl get secrets -A -o yaml`），响应包括所有秘密的内容。

Reviewer 3: ```suggestion
`kubectl get secrets -A -o yaml`），响应包含所有 Secret 的内容。
```

<!--
### Workload creation

Users who are able to create workloads (either Pods, or
[workload resources](/docs/concepts/workloads/controllers/) that manage Pods) will
be able to gain access to the underlying node unless restrictions based on the Kubernetes
[Pod Security Standards](/docs/concepts/security/pod-security-standards/) are in place.
-->
### 工作负载的创建 {#workload-creation}

> 能够创建工作负载的用户（Pod 或管理 Pod 的

Reviewer 3: ```suggestion
能够创建工作负载的用户（Pod 或管理 Pod 的[工作负载资源](/zh/docs/concepts/workloads/controllers/)）
```

> [工作负载资源](/zh/docs/concepts/workloads/controllers/)） 

Reviewer 3: ```suggestion
```

能够访问底层节点，除非基于 Kubernetes 的
[Pod 安全标准](/zh/docs/concepts/security/pod-security-standards/)做限制。

<!--
Users who can run privileged Pods can use that access to gain node access and potentially to
further elevate their privileges. Where you do not fully trust a user or other principal
with the ability to create suitably secure and isolated Pods, you should enforce either the
**Baseline** or **Restricted** Pod Security Standard.
You can use [Pod Security admission](/docs/concepts/security/pod-security-admission/)
or other (third party) mechanisms to implement that enforcement.
-->
可以运行特权 Pod 的用户可以利用该访问权限获得节点访问权限，
并可能进一步提升他们的特权。如果你不完全信任具有创建适当安全和隔离 
Pod 的能力的用户或其他委托人，你应该按照 **Baseline**

> 或 **Restricted** Pod 安全标准强制执行。

Reviewer 3: ```suggestion
或 **Restricted** Pod 安全标准强制执行。你可以使用
```

> 你可以使用 

Reviewer 3: ```suggestion
```

> [Pod 安全性准入](/zh/docs/concepts/security/pod-security-admission/)

Reviewer 3: ```suggestion
[Pod 安全性准入](/zh/docs/concepts/security/pod-security-admission/)或其他（第三方）机制来实施该执行。
```

> 或其他（第三方）机制来实施该执行。

Reviewer 3: ```suggestion
```

<!--
You can also use the deprecated [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/) mechanism
to restrict users' abilities to create privileged Pods (N.B. PodSecurityPolicy is scheduled for removal
in version 1.25).
-->

> 你还可以使用已弃用的 [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/)

Reviewer 3: ```suggestion
你还可以使用已弃用的 [PodSecurityPolicy](/zh/docs/concepts/policy/pod-security-policy/)
```

Author: `zh/docs/concepts/security/pod-security-policy/` this one has not been translated yet, shall we use the English link first?

Reviewer 2: It is a dilemma. I'd suggest we use the '/zh` prefix here and translate that page asap.

> 机制以限制用户创建特权 Pod 的能力 （特别注意： PodSecurityPolicy 已计划在版本 1.25 中删除）。

Reviewer 3: ```suggestion
机制以限制用户创建特权 Pod 的能力（特别注意：PodSecurityPolicy 已计划在版本 1.25 中删除）。
```

Author: 这里属于中英文混排，是否保留一个空格？

Reviewer 3: 这里是标点符号和英文混合的情况，不需要增加空格。我个人的翻译经验是这样：
- 中英文混合，在英文或者数字的前后各增加一个空格，英文处在行首或者行尾则不需要
- 中文和数字混合，同上
- 标点符合和英文混合，英文前后都不需要增加空格

换行会自动添加空格，所以长句会优先选择在中英文交接处断句，其次是标点符合

<!--
Creating a workload in a namespace also grants indirect access to Secrets in that namespace. 
Creating a pod in kube-system or a similarly privileged namespace can grant a user access to 
Secrets they would not have through RBAC directly.
-->
在命名空间中创建工作负载还会授予对该命名空间中 Secret 的间接访问权限。

> 在 kube-system 或类似特权的命名空间中创建 pod 

Reviewer 3: ```suggestion
在 kube-system 或类似特权的命名空间中创建 Pod
```

可以授予用户不需要通过 RBAC 即可获取 Secret 访问权限。

<!--
### Persistent volume creation

As noted in the [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems) documentation, access to create PersistentVolumes can allow for escalation of access to the underlying host. Where access to persistent storage is required trusted administrators should create 
PersistentVolumes, and constrained users should use PersistentVolumeClaims to access that storage.
-->
### 持久卷的创建 {#persistent-volume-creation}

> 如 [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems) 

Reviewer 3: ```suggestion
如 [PodSecurityPolicy](/zh/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems)
```

Author: This one has not been translated yet。 😄

Reviewer 3: yeah, since localization can not keep translated in a DAG pattern, 😆 
so let's translate this as soon as possible

文档中所述，创建 PersistentVolumes 的权限可以提权访问底层主机。

> 如果需要访问 PersistentVolume，受信任的管理员创建 `PersistentVolume`，

Reviewer 3: ```suggestion
如果需要访问 PersistentVolume，受信任的管理员应该创建 `PersistentVolume`，
```

> 受约束的用户使用 `PersistentVolumeClaim` 访问该存储。

Reviewer 3: ```suggestion
受约束的用户应该使用 `PersistentVolumeClaim` 访问该存储。
```

<!--
### Access to `proxy` subresource of Nodes

Users with access to the proxy sub-resource of node objects have rights to the Kubelet API, 
which allows for command execution on every pod on the node(s) which they have rights to. 
This access bypasses audit logging and admission control, so care should be taken before 
granting rights to this resource.
-->

> ### 访问节点的“代理”子资源  {#access-to-proxy-subresource-of-nodes}

Reviewer 3: ```suggestion
### 访问节点的 `proxy` 子资源  {#access-to-proxy-subresource-of-nodes}
```

where the signer is `kubernetes.io/kube-apiserver-client` to create new client certificates 
which allow users to authenticate to the cluster. Those client certificates can have arbitrary 
names including duplicates of Kubernetes system components. This will effectively allow for privilege escalation.
-->
### CSR 和证书颁发 {#csrs-and-certificate-issuing}

CSR API 允许用户拥有 `create` CSR 的权限和 `update`
`certificatesigningrequests/approval` 的权限，
其中签名者是 `kubernetes.io/kube-apiserver-client`,
通过此签名创建的客户端证书允许用户向集群进行身份验证。
这些客户端证书可以包含任意的名称，包括 Kubernetes 系统组件的副本。
这将有利于特权升级。

<!--
### Token request

Users with `create` rights on `serviceaccounts/token` can create TokenRequests to issue 
tokens for existing service accounts. 
-->
### 令牌请求 {#token-request}

拥有 `serviceaccounts/token` 的 `create` 权限的用户可以创建 
TokenRequests 来发布现有服务帐户的令牌。

<!--
### Control admission webhooks

Users with control over `validatingwebhookconfigurations` or `mutatingwebhookconfigurations` 
can control webhooks that can read any object admitted to the cluster, and in the case of 
mutating webhooks, also mutate admitted objects.
-->
###  控制准入 Webhook {#control-admission-webhooks}

> 可以操作 `validatingwebhookconfigurations` 或 `mutatingwebhookconfigurations`

Reviewer 3: ```suggestion
可以控制 `validatingwebhookconfigurations` 或 `mutatingwebhookconfigurations`
```

Author: 操作是否更合适下？

Reviewer 2: 控制好些

> 的用户可以操作控制 Webhook 和变更 Webhook，控制 Webhook 可以读取任何准入集群的对象，

Reviewer 3: ```suggestion
的用户可以控制能读取任何允许进入集群的对象的 webhook，
```

> 变更 Webhook 可以变更准入的对象。

Reviewer 3: ```suggestion
并且在有变更 webhook 的情况下，还可以变更准入的对象。
```

<!--
## Kubernetes RBAC - denial of service risks {#denial-of-service-risks}

### Object creation denial-of-service {#object-creation-dos}
Users who have rights to create objects in a cluster may be able to create sufficient large 
objects to create a denial of service condition either based on the size or number of objects, as discussed in
[etcd used by Kubernetes is vulnerable to OOM attack](https://github.com/kubernetes/kubernetes/issues/107325). This may be
specifically relevant in multi-tenant clusters if semi-trusted or untrusted users 
are allowed limited access to a system.

One option for mitigation of this issue would be to use [resource quotas](/docs/concepts/policy/resource-quotas/#object-count-quota)
to limit the quantity of objects which can be created.
-->
## Kubernetes RBAC - 拒绝服务攻击的风险 {#denial-of-service-risks}

### 对象创建拒绝服务 {#object-creation-dos}

> 有权在集群中创建对象的用户根据创建对象的大小和数量可能会创建足够大的对象,

Reviewer 3: ```suggestion
有权在集群中创建对象的用户根据创建对象的大小和数量可能会创建足够大的对象，
```

创造出拒绝服务攻击的条件，如
[Kubernetes 使用的 etcd 容易受到 OOM 攻击]（https://github.com/kubernetes/kubernetes/issues/107325）
中的讨论。允许太不受信任或者不受信任的用户对系统进行有限的访问在多租户集群中是特别重要的。

> 缓解此问题的一种选择是使用

Reviewer 3: 合并这三行

---

> reviewers:

Reviewer: Remove `reviewers` metadata, even if it's empty.

> title: 基于角色的访问控制良好实践

Reviewer 4: ```suggestion
title: 基于角色的访问控制最佳实践
```

Author: If the title is 'best practice', should be translated "最佳实践". When I was translating, I also considered whether it would be better to translate it as "最佳实践" or “良好实践”, but in the end, it was better to keep the English meaning and think it should be translated as “良好实践”.  😄

description: >
  为集群操作人员提供的良好的 RBAC 设计原则和实践。
content_type: concept
---

<!--
---
reviewers:
title: Role Based Access Control Good Practices
description: >
  Principles and practices for good RBAC design for cluster operators.
content_type: concept
---

> 

Reviewer: Remove this two lines.

-->

<!-- overview -->

<!--
Kubernetes {{< glossary_tooltip text="RBAC" term_id="rbac" >}} is a key security control 
to ensure that cluster users and workloads have only the access to resources required to 
execute their roles. It is important to ensure that, when designing permissions for cluster
users, the cluster administrator understands the areas where privilge escalation could occur, 
to reduce the risk of excessive access leading to security incidents.

The good practices laid out here should be read in conjunction with the general [RBAC documentation](/docs/reference/access-authn-authz/rbac/#restrictions-on-role-creation-or-update).
-->

Kubernetes {{< glossary_tooltip text="RBAC" term_id="rbac" >}}
是保证集群用户和工作负载只能访问自身角色所需资源的关键安全控制器。

> 在为群集设计权限时，请务必确保集群集管理员知道可能发生特权升级的区域，

Reviewer: ```suggestion
在为集群设计权限时，请务必确保集群管理员知道可能发生特权升级的区域，
```

降低因过度访问导致安全事件的风险。

此文档的良好实践应该与
[RBAC 文档](/zh/docs/reference/access-authn-authz/rbac/#restrictions-on-role-creation-or-update)
一起阅读。

<!-- body -->

<!--
## General good practice

### Least privilege
-->

> ## 一般好的做法 {#general-good-practice}

Reviewer: ```suggestion
## 通用的最佳實踐 {#general-good-practice}
```

> ### 最小特权  {#least-privilege}

Reviewer: ```suggestion
### 最小權限  {#least-privilege}
```

   not just to all object types presently in the cluster, but also to all future object types
   which are created in the future.
 - Administrators should not use `cluster-admin` accounts except where specifically needed. 
   Providing a low privileged account with [impersonation rights](/docs/reference/access-authn-authz/authentication/#user-impersonation)
   can avoid accidental modification of cluster resources.
 - Avoid adding users to the `system:masters` group. Any user who is a member of this group 
   bypasses all RBAC rights checks and will always have unrestricted superuser access, which cannot be 
   revoked by removing Role Bindings or Cluster Role Bindings. As an aside, if a cluster is 
   using an authorization webhook, membership of this group also bypasses that webhook (requests 
   from users who are members of that group are never sent to the webhook)
-->
- 尽可能在命名空间级别分配权限。授予用户在特定命名空间中的权限时使用 RoleBinding
  而不是 ClusterRoleBinding。
- 尽可能避免提供通配符权限，尤其是对所有资源的权限。
  由于 Kubernetes 是一个可扩展的系统，因此提供通配符访问权限不仅会授予集群中当前的所有对象类型，
  还包含所有未来的将被创建的所有对象类型。
- 管理员不应使用 `cluster-admin` 账号，除非特别需要。为低特权帐户提供
  [模拟权限](/docs/reference/access-authn-authz/authentication/#user-impersonation)

>   可以避免意外修改群集资源。

Reviewer: ```suggestion
  可以避免意外修改集群资源。
```

 
<!--
### Minimize distribution of privileged tokens
-->
### 最大限度地减少特权令牌的分发 {#minimize-distribution-of-privileged-tokens}

<!--
Ideally, pods shouldn't be assigned service accounts that have been granted powerful permissions (for example, any of the rights listed under
[privilege escalation risks](#privilege-escalation-risks)). 
In cases where a workload requires powerful permissions, consider the following practices:
 - Limit the number of nodes running powerful pods. Ensure that any DaemonSets you run
  are necessary and are run with least privilege to limit the blast radius of container escapes.
 - Avoid running powerful pods alongside untrusted or publicly-exposed ones. Consider using 
   [Taints and Toleration](/docs/concepts/scheduling-eviction/taint-and-toleration/), [NodeAffinity](/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity), or [PodAntiAffinity](/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) to ensure 
   pods don't run alongside untrusted or less-trusted Pods. Pay especial attention to
   situations where less-trustworthy Pods are not meeting the **Restricted** Pod Security Standard.
-->

> 理想情况下，不应为 Pod 分配授予强大权限的服务帐户,（例如，在

Reviewer: ```suggestion
理想情况下，不应为 Pod 分配授予强大权限的服务帐户，（例如，在
```

[特权提权的风险](#privilege-escalation-risks)中列出的任一权限）。
如果工作负载需要比较大的权限，请考虑以下做法：

> - 限制运行此类 Pod 的节点数量。确保您运行的任何 DaemonSet 都是必需的，

Reviewer: ```suggestion
- 限制运行此类 Pod 的节点数量。确保你运行的任何 DaemonSet 都是必需的，
```

  此参数可覆盖 Pod 服务账号设置，而需要服务账号令牌的工作负载仍可以挂载。

<!--
### Periodic review

It is vital to periodically review the Kubernetes RBAC settings for redundant entries and 
possible privilege escalations.
If an attacker is able to create a user account with the same name as a deleted user,
they can automatically inherit all the rights of the deleted user, especially the
rights assigned to that user.
 --> 
### 定期检查  {#periodic-review}
定期检查 Kubernetes RBAC 设置是否有冗余条目和提权可能性是至关重要的。
如果攻击者能够创建与已删除用户同名的用户账号，
他们可以自动继承被删除用户的所有权限，尤其是分配给该用户的权限。

<!--
## Kubernetes RBAC - privilege escalation risks {#privilege-escalation-risks}

Within Kubernetes RBAC there are a number of privileges which, if granted, can allow a user or a service account
to escalate their privileges in the cluster or affect systems outside the cluster.

This section is intended to provide visibility of the areas where cluster operators 
should take care, to ensure that they do not inadvertantly allow for more access to clusters than intended.
-->
## Kubernetes RBAC - 特权提权的风险 {#privilege-escalation-risks}

在 Kubernetes RBAC 中有许多特权，如果被授予，

> 用户或服务帐户可以提升其在群集中的权限并可能影响集群外的系统。

Reviewer: ```suggestion
用户或服务帐户可以提升其在集群中的权限并可能影响集群外的系统。
```

本节旨在为集群操作员提供应注意所在区域的可见性，
以确保它们不会无意中允许对集群的访问超出预期。

<!--
### Listing secrets

It is generally clear that allowing `get` access on Secrets will allow a user to read their contents.
It is also important to note that `list` and `watch` access also effectively allow for users to reveal the Secret contents.
For example, when a List response is returned (for example, via `kubectl get secrets -A -o yaml`), the response
includes the contents of all Secrets.
-->
### 列举 Secret {#listing-secrets}

通常很清楚，若设置 Secrets 的 `get` 访问权限，

> 用户可以获取其内容。 同样重要需要注意的是： `list` 和

Reviewer: ```suggestion
用户可以获取其内容。同样重要需要注意的是：`list` 和
```

`watch` 访问也允许用户获取 Secret 的内容。

> 例如，当返回 List 响应时（如,通过 

Reviewer: ```suggestion
例如，当返回 List 响应时（例如透過
```

Author: I think "如" and "例如" are the same means.

Reviewer: Yeah, it doesn't matter to me. But why you use `,` instead of `，` in Mandarin?

`kubectl get secrets -A -o yaml`），响应包括所有秘密的内容。

<!--
### Workload creation

Users who are able to create workloads (either Pods, or
[workload resources](/docs/concepts/workloads/controllers/) that manage Pods) will
be able to gain access to the underlying node unless restrictions based on the Kubernetes
[Pod Security Standards](/docs/concepts/security/pod-security-standards/) are in place.
-->
### 工作负载的创建 {#workload-creation}

> 能够创建工作负载的用户（Pod 或 管理 Pod 的

Reviewer: ```suggestion
能够创建工作负载的用户（Pod 或管理 Pod 的
```

[工作负载资源](/zh/docs/concepts/workloads/controllers/)） 
能够访问底层节点，除非基于 Kubernetes 的
[Pod 安全标准](/zh/docs/concepts/security/pod-security-standards/)做限制。

<!--
Users who can run privileged Pods can use that access to gain node access and potentially to
further elevate their privileges. Where you do not fully trust a user or other principal
with the ability to create suitably secure and isolated Pods, you should enforce either the
**Baseline** or **Restricted** Pod Security Standard.
You can use [Pod Security admission](/docs/concepts/security/pod-security-admission/)
or other (third party) mechanisms to implement that enforcement.
-->
可以运行特权 Pod 的用户可以利用该访问权限获得节点访问权限，
并可能进一步提升他们的特权。如果你不完全信任具有创建适当安全和隔离 
Pod 的能力的用户或其他委托人，你应该按照 **Baseline**
或 **Restricted** Pod 安全标准强制执行。

> 您可以使用 

Reviewer: ```suggestion
你可以使用
```

[Pod 安全性准入](/zh/docs/concepts/security/pod-security-admission/)
或其他（第三方）机制来实施该执行。

<!--
You can also use the deprecated [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/) mechanism
to restrict users' abilities to create privileged Pods (N.B. PodSecurityPolicy is scheduled for removal
in version 1.25).
-->

> 您还可以使用已弃用的 [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/)

Reviewer: ```suggestion
你还可以使用已弃用的 [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/)
```

> 机制以限制用户创建特权 Pod 的能力 （特别注意, PodSecurityPolicy 已计划在版本 1.25 中删除）。

Reviewer: ```suggestion
机制以限制用户创建特权 Pod 的能力 （特别注意：PodSecurityPolicy 已计划在版本 1.25 中删除）。
```

Author: nice catch, thanks。 I'm just wondering if a comma is better or a colon

<!--
Creating a workload in a namespace also grants indirect access to Secrets in that namespace. 
Creating a pod in kube-system or a similarly privileged namespace can grant a user access to 
Secrets they would not have through RBAC directly.
-->
在命名空间中创建工作负载还会授予对该命名空间中 Secret 的间接访问权限。
在 kube-system 或类似特权的命名空间中创建 pod 
可以授予用户不需要通过 RBAC 即可获取 Secret 访问权限。

<!--
### Persistent volume creation

As noted in the [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems) documentation, access to create PersistentVolumes can allow for escalation of access to the underlying host. Where access to persistent storage is required trusted administrators should create 
PersistentVolumes, and constrained users should use PersistentVolumeClaims to access that storage.
-->
### 持久卷的创建 {#persistent-volume-creation}

如 [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems) 
文档中所述，创建 PersistentVolumes 的权限可以提权访问底层主机。

> 如果需要访问 PersistentVolume，受信任的管理员创建 `PersistentVolume`
> ，受约束的用户使用 `PersistentVolumeClaim` 访问该存储。

Reviewer: ```suggestion
如果需要访问 PersistentVolume，受信任的管理员创建 `PersistentVolume`，
受约束的用户使用 `PersistentVolumeClaim` 访问该存储。
```

mutating webhooks, also mutate admitted objects.
-->
###  控制准入 Webhook {#control-admission-webhooks}

可以操作 `validatingwebhookconfigurations` 或 `mutatingwebhookconfigurations`
的用户可以操作控制 Webhook 和变更 Webhook，控制 Webhook 可以读取任何准入集群的对象，
变更 Webhook 可以变更准入的对象。

<!--
## Kubernetes RBAC - denial of service risks {#denial-of-service-risks}

### Object creation denial-of-service {#object-creation-dos}
Users who have rights to create objects in a cluster may be able to create sufficient large 
objects to create a denial of service condition either based on the size or number of objects, as discussed in
[etcd used by Kubernetes is vulnerable to OOM attack](https://github.com/kubernetes/kubernetes/issues/107325). This may be
specifically relevant in multi-tenant clusters if semi-trusted or untrusted users 
are allowed limited access to a system.

One option for mitigation of this issue would be to use [resource quotas](/docs/concepts/policy/resource-quotas/#object-count-quota)
to limit the quantity of objects which can be created.
-->
## Kubernetes RBAC - 拒绝服务攻击的风险 {#denial-of-service-risks}

### 对象创建拒绝服务 {#object-creation-dos}
有权在集群中创建对象的用户根据创建对象的大小和数量可能会创建足够大的对象,
创造出拒绝服务攻击的条件，如

> [Kubernetes 使用的 etcd 容易受到 OOM 攻击]（https://github.com/kubernetes/kubernetes/issues/107325）

Reviewer: ```suggestion
[Kubernetes 使用的 etcd 容易受到 OOM 攻击](https://github.com/kubernetes/kubernetes/issues/107325)
```
